### PR TITLE
Add: filter by category, latest and searching

### PR DESCRIPTION
**Pull request by saurabhchirde**
Features Added

1. Filter video by category
2. Search a Video.

filter-search

### CODE REVIEW

File: src/Utils/filterByLatest.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
+export const filterByLatest = (filteredArray, byLatest) => {
+  const withouFiltered = [...filteredArray];
+  if (byLatest === "") {
+    return withouFiltered;
```
Comment (m26unkwn) [inline src/Utils/filterByLatest.js:4]: why you are writing this if condition?

**Review by m26unkwn — APPROVED** [review]
Everything is good and I like the way you are handling the filtration.
You can check the given comment 
otherwise everything is good to go.

File: src/Pages/VideoListing/VideoListing.jsx [hunk 2]
```diff
@@ -1,31 +1,110 @@
...
   const {
     videoState: { videos, categories },
   } = useVideo();
+  const { filterState, filterDispatch } = useFilter();
+  const { byCategory, bySearch, byLatest } = filterState;
+  const { search } = useLocation();
+  const urlParam = new URLSearchParams(search);
+  const searchQuery = urlParam.get("query");
+
+  const onCategoryClickHandler = (category) => {
```
Comment (ritikaawasthy) [inline src/Pages/VideoListing/VideoListing.jsx:19]: Just a suggestion:  you could omit "on" just make it categoryClickHandler and for other functions as well?

File: src/Pages/VideoListing/VideoListing.jsx [hunk 3]
```diff
@@ -1,31 +1,110 @@
...
+    }
+  }, [searchQuery]);
 
   return (
     <div className="video-listing-body">
       <div className="discover-section"></div>
       <form className="category-section">
-        <div className="category-label">All</div>
+        <div
+          onClick={onAllClickHandler}
```
Comment (ritikaawasthy) [inline src/Pages/VideoListing/VideoListing.jsx:52]: should'nt this be a callback function in onClick?

File: src/Utils/filterByLatest.js (hunk 1 above)
Comment (ritikaawasthy) [reply src/Utils/filterByLatest.js:4]: Agreed this could be omitted

File: src/Utils/filterByLatest.js [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+  } else {
+    withouFiltered.sort((a, b) => {
+      let date1 = a.snippet.publishedAt.slice(0, 10).split("-").join("");
```
Comment (ritikaawasthy) [inline src/Utils/filterByLatest.js:7]: you could extract date using a function instead of using split,slice,join

**Review by ritikaawasthy — COMMENTED** [review]
Your code is structured well, theres good uniformity in naming as well. Just added some suggestion you could look at. otherwise looks great.

File: src/Utils/filterByLatest.js (hunk 1 above)
Comment (saurabhchirde) [reply src/Utils/filterByLatest.js:4]: Ok will fix this

File: src/Pages/VideoListing/VideoListing.jsx (hunk 3 above)
Comment (saurabhchirde) [reply src/Pages/VideoListing/VideoListing.jsx:52]: If we are calling a declared function then we provide that function name. If we don't want to declare a function then we provide a callback function.

File: src/Utils/filterByLatest.js (hunk 4 above)
Comment (saurabhchirde) [reply src/Utils/filterByLatest.js:7]: Noted, will fix this.

**Pull request merged**